storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/ConstraintSolverStrategy.java

```diff
@@ -320,51 +320,101 @@ private boolean checkSchedulingFeasibility(int maxStateSearch) {
 
-                    SolverResult results = backtrackSearch(state.nextExecutor());
-                    if (results.success) {
-                        //We found a good result we are done.
-                        return results;
+            int progressIdx = -1;
+            boolean assigned = false;
+            for (String nodeId : sortedNodesIter) {
+                RasNode node = nodes.get(nodeId);
+                for (WorkerSlot workerSlot : node.getSlotsAvailableToScheduleOn()) {
+                    progressIdx++;
+                    if (progressIdx <= progressIdxForExec[execIndex]) {
+                        continue;
                     }
+                    progressIdxForExec[execIndex]++;
+                    LOG.debug("backtrackSearch: loopCnt = {}, state.execIndex = {}, node/slot-ordinal = {}, nodeId = {}",
+                        loopCnt, execIndex, progressIdx, nodeId);
 
-                    if (state.areSearchLimitsExceeded()) {
-                        //No need to search more it is not going to help.
-                        return new SolverResult(state, false);
+                    if (!isExecAssignmentToWorkerValid(workerSlot, state)) {
+                        continue;
                     }
 
-                    //backtracking (If we ever get here there really isn't a lot of hope that we will find a scheduling)
-                    state.backtrack(execToComp, node, workerSlot);
+                    try {
+                        state.incStatesSearched();
+                        state.tryToSchedule(execToComp, node, workerSlot);
+                        if (state.areAllExecsScheduled()) {
+                            //Everything is scheduled correctly, so no need to search any more.
+                            LOG.info("backtrackSearch: AllExecsScheduled at loopCnt = {} in {} milliseconds, elapsedtime in state={}",
+                                loopCnt, System.currentTimeMillis() - startTimeMilli, Time.currentTimeMillis() - state.startTimeMillis);
+                            return new SolverResult(state, true);
+                        }
+                        state = state.nextExecutor();
+                        nodeForExec[execIndex] = node;
+                        workerSlotForExec[execIndex] = workerSlot;
+                        assigned = true;
+                        LOG.debug("backtrackSearch: Assigned execId={} to node={}, node/slot-ordinal={} at loopCnt={}", 
+                            execIndex, nodeId, progressIdx, loopCnt);
+                        continue OUTERMOST_LOOP;
+                    } catch (Exception ex) {
+                        LOG.error("backtrackSearch: Failed to schedule execId {} to node/slot-ordinal = {} at loopCnt = {}, nodeId = {}",
+                            ex, execIndex, progressIdx, loopCnt, nodeId);
+                        continue;
+                    }
+                }
+            }
+            if (!assigned) {
```

Comment:
This seems redundant. `!assigned` will always be true if it ever reaches here

Reply:
assigned not required. will remove it.